scheduler/pkg/store/memory.go

```diff
@@ -375,7 +375,10 @@ func (m *MemoryStore) updateLoadedModelsImpl(
 	// this could be in the cases where we are scaling down a model and the new replica count can be all deployed
 	// and always send an update for deleted models, so the operator will remove them from k8s
 	// also send an update for progressing models so the operator can update the status in the case of a network glitch where the model generation has been updated
-	if replicaStateUpdated || modelVersion.state.State == ScheduleFailed || model.IsDeleted() || modelVersion.state.State == ModelProgressing {
+	// also send an update if the model is not yet at desired replicas, if we have partial scheduling
+
+	if replicaStateUpdated || modelVersion.state.State == ScheduleFailed || model.IsDeleted() || modelVersion.state.State == ModelProgressing ||
+		(modelVersion.state.State == ModelAvailable && len(modelVersion.GetAssignment()) < modelVersion.DesiredReplicas()) {
```

Comment:
Oof.. this statement is becoming unwieldy (if it wasn't already). I wonder if it's worth having an explicit task to figure out how to refactor it.

Reply:
Yeah we should fix it, it keeps growing.